tests/foreman/api/test_contentview.py

```diff
@@ -606,30 +546,23 @@ def test_positive_promote_2(self):
         @Feature: ContentView
 
         """
-        content_view = entities.ContentView(organization=self.org.id)
-        content_view.id = content_view.create_json()['id']
+        content_view = entities.ContentView(organization=self.org).create()
         content_view.set_repository_ids([self.yum_repo.id])
         content_view.publish()
+        content_view = content_view.read()
 
         # Promote the content view version.
-        cvv = entities.ContentViewVersion(
-            id=content_view.read_json()['versions'][0]['id']  # only one ver
-        )
         for _ in range(REPEAT):
-            lc_env_id = entities.LifecycleEnvironment(
-                organization=self.org.id
-            ).create_json()['id']
-            cvv.promote(lc_env_id)
+            lce = entities.LifecycleEnvironment(organization=self.org).create()
+            content_view.version[0].promote(lce.id)
 
         # Everything's done - check some content view attributes...
-        cv_attrs = content_view.read_json()
-        self.assertEqual(len(cv_attrs['repositories']), 1)
-        self.assertEqual(len(cv_attrs['versions']), 1)
+        content_view = content_view.read()
+        self.assertEqual(len(content_view.repository), 1)
+        self.assertEqual(len(content_view.version), 1)
 
         # ...and some content view version attributes.
-        cvv_attrs = entities.ContentViewVersion(
-            id=cv_attrs['versions'][0]['id']
-        ).read_json()
+        cvv_attrs = content_view.version[0].read_json()
```

Comment:
Is it specific reason to leave read_json() and two lines below here and not move to refactor procedure style?